src/services/azure-directory.service.ts

```diff
@@ -157,8 +147,15 @@ export class AzureDirectoryService extends BaseDirectoryService implements IDire
           if (!entry.deleted) {
             continue;
           }
-          if (await this.filterOutUserResult(setFilter, entry, false)) {
-            continue;
+
+          // filterOutUserResult should only be called if the filter is includeUser or excludeUser
```

Comment:
This just restates the `if` condition, I was thinking of something more like

```suggestion
          // Deleted users cannot be filtered by group in Azure AD so we only apply user filters if required
```

...assuming that is correct, that's my understanding of the rationale. Basically replacing the old comment on line 291 that we're now deleting.